### ISSUE TITLE
What is the meaning in ArcFace Loss formula?

### ISSUE THREAD

**Issue by vmthanh** [4uiiurz1/keras-arcface#19]
Hi. 
Thanks for your great implementation. Im trying to understand the mathematics behind the ArcFace in your implementation. 
But when I read to this line the **metrics.py**
` logits = logits * (1 - y) + target_logits * y`
I don't understand the link behind this line and the original paper. The code looks like the cross entropy loss before going to the Softmax function, but if if the cross entropy, should it have the log inside? AIm not sure about this part very much. Can you help me to understand it?

Many thanks

**Comment by SerChirag** [comment]
@4uiiurz1 I have the same query.

**Comment by ofathy1981** [comment]
can you please help me using this project with face dataset as input

**Comment by BRO-HAMMER** [comment]
@vmthanh I believe that line is assigning the value of cosine distance to the logits of the incorrect classes and cosine+margin to the logits of the correct one.
